dashboard/app/models/sections/section.rb

```diff
@@ -228,7 +228,11 @@ def update_student_sharing(sharing_disabled)
   def name_safe_students
     name_splitter_proc = ->(student) {FullNameSplitter.split(student.name)}
 
-    SafeNames.get_safe_names(students, name_splitter_proc).map do |safe_name_and_student|
+    students_only = students.filter do |user|
```

Comment:
I'm (very very slightly) concerned with this 'filter'... I forget (or just wholy ignorant of) the specific implications in rails 6. This is a relationship attribute, so it's a database cursor, yes? The filter may transform it to an array and then iterate on it... which with just a few dozen rows isn't awful, of course. That all said, it could be `students.where.not(user_type: "teacher")` to be more dogmatic and to ensure the logic is on the database engine.

Reply:
Naturally, the map later on already does this anyway so hopefully the filter is lazy, but I can't remember if that is the case. The `where` certainly is, though.

Reply:
Good callout. This highlights my Rails ignorance. I hadn't considered this was going to be a list that would then be individually queried. I'll try changing this with your suggestion.

Reply:
Rails is so tough to grok all at once and it's so easy to not see these that I was unsure until I tested it. It's a lovely subtle difference where we avoid an `N` on the ruby side (no real consequences in this case, tbh, since the N is, what, at most 100 or so): 
![image](https://github.com/code-dot-org/code-dot-org/assets/31674/200a4975-e4e0-4feb-928c-2df885bf41d7)

But what we get as a result is a database cursor instead of an array... so we just delay the execution of the query a little bit more and the filter work is being done on the database system, instead. I've developed a habit of looking for these opportunities and generally have the dogmatic stance of pushing query execution to as late as possible. Makes my head hurt sometimes tho. lol.

Reply:
Yep, makes perfect sense. Thanks for catching this!

Reply:
Changed from `.filter` to `.where` per the suggestion to avoid an N+1 query.